apps/ios/Shared/Views/UserSettings/PrivacySettings.swift

```diff
@@ -86,46 +87,45 @@ struct PrivacySettings: View {
 
                 Section {
                     settingsRow("person") {
-                        Toggle("Contacts", isOn: $sendReceiptsContacts)
+                        Toggle("Contacts", isOn: $contactReceipts)
                     }
-                    settingsRow("person.2") {
-                        Toggle("Small groups (max 10)", isOn: Binding.constant(false))
-                    }
-                    .foregroundColor(.secondary)
-                    .disabled(true)
+//                    settingsRow("person.2") {
+//                        Toggle("Small groups (max 20)", isOn: Binding.constant(false))
+//                    }
                 } header: {
                     Text("Send delivery receipts to")
                 } footer: {
                     VStack(alignment: .leading) {
                         Text("These settings are for your current profile **\(ChatModel.shared.currentUser?.displayName ?? "")**.")
-                        Text("They can be overridden in contact and group settings")
+                        Text("They can be overridden in contact settings")
                     }
                     .frame(maxWidth: .infinity, alignment: .leading)
                 }
-                .confirmationDialog("You have configured this setting for some contacts", isPresented: $showResetContactsReceiptsOverridesDialogue, titleVisibility: .visible) {
-                    Button("Cancel") {
-                        sendReceiptsContactsToggleReset = true
-                        sendReceiptsContacts.toggle()
+                .confirmationDialog(contactReceiptsDialogTitle, isPresented: $contactReceiptsDialogue, titleVisibility: .visible) {
+                    Button(contactReceipts ? "Enable (keep overrides)" : "Disable (keep overrides)") {
+                        setSendReceiptsContacts(contactReceipts, clearOverrides: false)
                     }
-                    Button("Keep per contact overrides") {
-                        setSendReceiptsContacts(sendReceiptsContacts, clearOverrides: false)
+                    Button(contactReceipts ? "Enable for all" : "Disable for all", role: .destructive) {
+                        setSendReceiptsContacts(contactReceipts, clearOverrides: true)
                     }
-                    Button("Reset per contact overrides", role: .destructive) {
-                        setSendReceiptsContacts(sendReceiptsContacts, clearOverrides: true)
+                    Button("Cancel", role: .cancel) {
+                        contactReceiptsReset = true
+                        contactReceipts.toggle()
                     }
                 }
             }
         }
         .onAppear {
-            if let currentUser = m.currentUser {
-                sendReceiptsContacts = currentUser.sendRcptsContacts
+            if let u = m.currentUser, contactReceipts != u.sendRcptsContacts {
+                contactReceiptsReset = true
```

Comment:
I've tried setting it as well and it sometimes lead to a different race where first toggle change wasn't working.. but maybe something else changed and it's ok now (it didn't bug again when I tested with new code)